### ISSUE TITLE
Allow enabling/disabling GitHub integration's PR comment feature per project

### ISSUE THREAD

**Issue by ento** [getsentry/sentry#76468]
### Problem Statement

We use a separate Sentry project for testing Sentry integration in local development environments so that local-only issues don't pollute the project that tracks issues in deployed environments. When someone has a misconfiguration or such in their local env that results in a Sentry issue, Sentry's GitHub integration sometimes adds a comment to the PR that might've caused the issue:

![Image](https://github.com/user-attachments/assets/7d6c228d-a799-4f43-b1fa-6473321d5a6f)

While this feature is generally useful, it becomes a noise when it's from the project for local testing, taking up people's time investigating what turns out to be a non-issue.

### Solution Brainstorm

Allow configuring these features project by project:
- Enable Comments on Suspect Pull Requests
- Enable Comments on Open Pull Requests

### Product Area

Settings - Integrations

**Comment by leedongwei** [comment]
Thanks! This is good feedback. I'll try to get it onto the roadmap.

**Comment by kpujjigit** [comment]
Hi All, any roadmap updates on this feature? This has been raised internally via EPD Hub.

**Comment by sentaur-athena** [comment]
@kpujjigit this is not on the roadmap yet. I'll post an update as soon as we have have a timeline.

**Comment by tim-gq** [comment]
I'm curious that this doesn't exist yet.  I was looking to disable them today for our staging environment.  Is Sentry's perspective that Sentry shouldn't really be used to capture issues in staging and non-productions environments?

**Comment by sentaur-athena** [comment]
This is not prioritized for near future roadmap. I'll update with timelines when we have one.

**Comment by ReneGreen27** [comment]
Related to internal tickets [149070](https://sentry.zendesk.com/agent/tickets/149070) and [148708](https://sentry.zendesk.com/agent/tickets/148708)

**Comment by tim-gq** [comment]
Don't know if it's possible, but another nice (or alternative) improvement would be to indicate on the notification that gets posted to github which environment(s) generated the error.

If it's production, I very likely want to drop what I'm doing and investigate.   If it's not production, my care level is much much lower.

**Comment by sentaur-athena** [comment]
Yes, that's a very good feedback. We're not working on improving this feature atm but let me create a separate ticket to track it in the backlog: https://github.com/getsentry/sentry/issues/88827

**Comment by tim-gq** [comment]
We got pretty close to turning it off too.  A couple of time's it's been highly valuable as the way to recognize a new issue in production though, so we decided to tolerate the noise.  Generates a groan each time we see it's a non-prod issue being reported though.

**Comment by sentaur-athena** [comment]
Adding this to the current cycle for a quick resolution -> adding environment to the comment.

**Comment by tim-gq** [comment]
@iamrajjoshi @sentaur-athena awesome to read this is completed.  How do I disable notifications for our staging projects?
